flexget/plugins/sites/wordpress.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import unicode_literals, division, absolute_import
+from builtins import *  # noqa pylint: disable=unused-import, redefined-builtin
+from future.moves.urllib.parse import urlencode
+
+import logging
+
+import re
+from flexget import plugin
+from flexget.event import event
+from flexget.plugin import PluginError
+from requests import Session
+from requests import Request, RequestException
+from requests.utils import dict_from_cookiejar, cookiejar_from_dict
+
+log = logging.getLogger('wordpress_auth')
+
+
+def construct_wp_login_request(url, username='', password='', redirect='/wp-admin/'):
+    headers = {
+        'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) '
+                      'Chrome/50.0.2661.102 Safari/537.36',
+        'Content-Type': 'application/x-www-form-urlencoded',
+        'DNT': '1'
+    }
+    data = {
+        'log': username,
+        'pwd': password,
+        'wp-submit': 'Log In',
+        'testcookie': '1',
+        'redirect_to': redirect
+    }
+    return Request(method='POST', url=url, headers=headers, data=urlencode(data).encode('UTF-8')).prepare()
+
+
+def match_wordpress_cookie(key):
+    return re.match(r'wordpress(?!_test)[A-z0-9]*', key, re.IGNORECASE)
+
+
+def send_request(session, prep_request, redirects=5):
```

Comment:
I'd honestly prefer having this inside `on_task_start`. You generally create a lot of global functions that could just as well be made inline in `on_task_start`.

I think the code becomes harder to read when having to scroll up and down to see what each function does. `construct_wp_login_request` is also just essentially two dicts.

It looks fine otherwise. I just don't agree with the code style.

Reply:
I've updated the code now so there are only 3 global functions. 
The request is now sent in `on_task_start`
All exceptions are now handled or raised in `on_task_start`